### PR TITLE
Test shard_context on already-created boost::asio::io_service.

### PR DESCRIPTION
**Pull request by dHannasch**
## Why are these changes needed?

I figured I'd model the test for PingPort() on the test for Connect(). But there doesn't seem to be a test for Connect(), so I'm trying to make one.
## Related issue number

https://github.com/ray-project/ray/pull/12022#issuecomment-732648281
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/test/asio_test.cc [hunk 1]
```diff
@@ -66,6 +67,13 @@ TEST_F(RedisAsioTest, TestRedisCommands) {
   redisAsyncCommand(ac, NULL, NULL, "SET key test");
   redisAsyncCommand(ac, GetCallback, nullptr, "GET key");
 
+  std::shared_ptr<RedisContext> shard_context =
+      std::make_shared<RedisContext>(io_service);
+  RAY_CHECK_OK(shard_context->Connect(std::string("127.0.0.1"),
```
Comment (richardliaw) [inline src/ray/gcs/test/asio_test.cc:72]: should we be using ASSERT_TRUE here instead?

cc @rkooo567

Comment (rkooo567) [reply src/ray/gcs/test/asio_test.cc:72]: Yeah let' use the ASSERT_TRUE

Comment (dHannasch) [reply src/ray/gcs/test/asio_test.cc]: Better?

**Review by richardliaw — APPROVED** [review]
This looks good to me! (that being said I am largely not involved in this part of the code) but this is only a test so its fine.

**Pull request merged**